src/components/Match/BuildingMap/BuildingMap.jsx

```diff
@@ -23,22 +36,148 @@ export default function BuildingMap({
+                    }}
+                  >
+                    {damage.map(player => (
+                      <div
+                        style={{
+                          width: `${(Number(player.damage) * 100) / buildingsHealth[type === 'tower' ? `tower${tier}` : type]}%`,
+                          backgroundColor: playerColors[player.player_slot],
+                        }}
+                      />
+                    ))}
+                  </div>
+                  <div className={styles.damage}>
+                    {damage.map(player => (
+                      <div>
+                        <img
+                          src={heroes[player.hero_id] && API_HOST + heroes[player.hero_id].img}
+                          role="presentation"
+                        />
+                        <span className={styles.damageValue}>
+                          {player.damage}
+                        </span>
+                        <span
+                          style={{ color: playerColors[player.player_slot] }}
+                          className={styles.playerName}
+                        >
+                          {player.name}
+                        </span>
+                        {destroyedBy && destroyedBy.player_slot === player.player_slot &&
+                          <span className={styles.lasthit}>
+                            {type !== 'fort' && <img src={`${API_HOST}/apps/dota2/images/tooltips/gold.png`} role="presentation" />}
+                            got a lasthit
+                          </span>
+                        }
+                      </div>
+                    ))}
+                    {(damageByCreeps > 0) && (bits[i] !== '1') &&
+                      <div className={styles.creeps}>
+                        <img
+                          src="/assets/images/blank-1x1.gif"
+                          role="presentation"
+                          style={{ backgroundColor: side === 'good' ? styles.red : styles.green }}
+                        />
+                        <span className={styles.damageValue}>
+                          {damageByCreeps}
+                        </span>
+                        <span
+                          style={{ color: side === 'good' ? styles.red : styles.green }}
+                          className={styles.playerName}
+                        >
+                          creeps {!destroyedBy &&
+                            <span className={styles.lasthit}>
+                              got a lasthit
+                            </span>
+                          }
+                        </span>
+                      </div>
+                    }
+                    {type.includes('rax') && damageByCreeps < 0 &&
+                      <div className={styles.regenerated}>
+                        Regenerated {Math.abs(damageByCreeps)} hit points
```

Comment:
You are right, but I was thinking its ok to display approximate value if it possible. I can remove it.